src/plugins/core/data_validation.ts

```diff
@@ -46,6 +46,34 @@ export class DataValidationPlugin
     for (const sheetId of sheetIds) {
       this.loopThroughRangesOfSheet(sheetId, applyChange);
     }
+    this.loopThroughAllRanges(applyChange);
+  }
+
+  private loopThroughAllRanges(applyChange: ApplyRangeChange) {
+    for (const sheetId of Object.keys(this.rules)) {
+      const rules = this.rules[sheetId];
+      for (let ruleIndex = rules.length - 1; ruleIndex >= 0; ruleIndex--) {
+        const rule = this.rules[sheetId][ruleIndex];
+        for (let valueIdx = 0; valueIdx < rule.criterion.values.length; valueIdx++) {
+          let value = this.getters.adaptFormulaStringDependencies(
+            sheetId,
+            rule.criterion.values[valueIdx],
+            applyChange
+          );
+          if (value != rule.criterion.values[valueIdx]) {
```

Comment:
this `if` is not strictly necessary. The history is not changed if the value didn't